### PR TITLE
dftd3: make dftd3 harness-like

### PR DESCRIPTION
**Pull request by loriab**
ok use of `extras['local_keywords']`?

### CODE REVIEW

File: qcengine/programs/dftd3.py [hunk 1]
```diff
@@ -62,326 +61,224 @@ def get_version(self) -> str:
...
-    modulerec['env'] = {
-            'HOME': os.environ.get('HOME'),
-            'PATH': os.pathsep.join([os.path.abspath(x) for x in os.environ.get('PSIPATH', '').split(os.pathsep) if x != '']) + \
-                    os.pathsep + os.environ.get('PATH'),
-            'LD_LIBRARY_PATH': os.environ.get('LD_LIBRARY_PATH'),
+        return {
+            "command": command,
+            "infiles": infiles,
+            "outfiles": ['dftd3_gradient', 'dftd3_abc_gradient'],
+            "scratch_location": config.scratch_directory,
```
Comment (dgasmith) [inline qcengine/programs/dftd3.py]: We should standardize location/directory one way or the other across the whole code base. I vote a bit for `scratch_directory` as this keyword is found in the large QCA ecosystem.

File: qcengine/programs/dftd3.py [hunk 2]
```diff
@@ -62,326 +61,224 @@ def get_version(self) -> str:
...
+
+        retres = calcinfo[f'CURRENT {input_model.driver.upper()}']
+        if isinstance(retres, Decimal):
+            retres = float(retres)
+        elif isinstance(retres, np.ndarray):
+            retres = retres.ravel().tolist()
+
+        output_data = {
+            'extras': {
+                'local_keywords': input_model.extras['info'],
```
Comment (dgasmith) [inline qcengine/programs/dftd3.py]: No restrictions on extras, the present info seems reasonable however.

File: qcengine/programs/dftd3.py (hunk 1 above)
Comment (loriab) [reply qcengine/programs/dftd3.py]: If it meant the place directly where the calc happens, I would agree. But `scratch_location` turns into `parent` https://github.com/MolSSI/QCEngine/blob/master/qcengine/util.py#L418 or "scratch directory directory." Is that what the generic QCA `scratch_directory` means?

File: qcengine/programs/dftd3.py (hunk 2 above)
Comment (loriab) [reply qcengine/programs/dftd3.py]: `local_keywords` is sort of "resolved keywords" and is the best record of what the calc did. I don't really care about the name of the key. Not following "No restrictions on extras."

Comment (loriab) [reply qcengine/programs/dftd3.py]: nm, I had forgotten I asked the question at the top, so was trying to figure out the point of your comment.

File: qcengine/programs/dftd3.py (hunk 1 above)
Comment (dgasmith) [reply qcengine/programs/dftd3.py]: I mean from a QCA perspective it doesn't matter. It is the location to spawn tasks, if a further tmpdir is built it is quite irrelevant as the tmpdirs are always cleaned. There could be additional complications that I am not considering however.

Comment (loriab) [reply qcengine/programs/dftd3.py]: I guess I use “scratch directory” two ways, too. The enclosing loc like /scratch/hohndoe/ or the qc run loc like /tmp/water.4444/. Along that like, I don’t really care directory or location. Can change.

**Comment by loriab** [comment]
ready for review

**Pull request merged**